File: src/simple.js
import { validateConfig } from './config.js';
import { createModels, callModelsInParallel } from './models/index.js';
import { ConversationManager } from './conversation.js';

async function main() {
  console.log('Multi-Model AI System - Simple Test');
  console.log('===================================\n');
  
  try {
    validateConfig();
  } catch (error) {
    console.error('Configuration error:', error.message);
    process.exit(1);
  }

  const models = createModels();
  const conversation = new ConversationManager('test-conversation.json');
  
  // Test with a simple prompt
  const prompt = 'Compare the number systems used in ancient Rome and ancient Egypt.';
  console.log(`Prompt: ${prompt}\n`);
  console.log('Sending to all models...\n');
  
  const messages = [{ role: 'user', content: prompt }];
  const responses = await callModelsInParallel(models, messages);
  
  for (const [model, response] of Object.entries(responses)) {
    console.log(`\n=== ${model} Response ===`);
    if (response.error) {
      console.log(`Error: ${response.error}`);
    } else {
      console.log(response.content);
      if (response.usage?.total_tokens) {
        console.log(`\nTokens: ${response.usage.total_tokens}`);
      }
    }
  }
  
  // Save the conversation
  conversation.addRound(prompt, responses);
  await conversation.save();
  console.log('\n\nConversation saved to test-conversation.json');
  
  // Test assessment
  console.log('\n\n--- Running Assessment ---\n');
  
  const assessmentPrompt = conversation.formatAssessmentPrompt(prompt, responses);
  const assessmentMessages = [{ role: 'user', content: assessmentPrompt }];
  
  const assessmentResponses = await callModelsInParallel(models, assessmentMessages);
  
  for (const [model, response] of Object.entries(assessmentResponses)) {
    console.log(`\n=== ${model} Assessment ===`);
    if (response.error) {
      console.log(`Error: ${response.error}`);
    } else {
      console.log(response.content.substring(0, 500) + '...');
    }
  }
  
  conversation.addRound(assessmentPrompt, assessmentResponses, true);
  await conversation.save();
}

main().catch(console.error);

File: archive/src_python/simple.py
#!/usr/bin/env python3
import asyncio
import sys
from .config import validate_config
from .models.index import create_models, call_models_in_parallel
from .conversation import ConversationManager


async def main():
    print('Multi-Model AI System - Simple Test')
    print('===================================\n')
    
    try:
        validate_config()
    except ValueError as e:
        print(f'Configuration error: {e}')
        sys.exit(1)
    
    models = create_models()
    conversation = ConversationManager('test-conversation.json')
    
    # Test with a simple prompt
    prompt = 'Compare the number systems used in ancient Rome and ancient Egypt.'
    print(f'Prompt: {prompt}\n')
    print('Sending to all models...\n')
    
    messages = [{'role': 'user', 'content': prompt}]
    responses = await call_models_in_parallel(models, messages)
    
    for model, response in responses.items():
        print(f'\n=== {model} Response ===')
        if 'error' in response:
            print(f'Error: {response["error"]}')
        else:
            print(response['content'])
            if response.get('usage') and response['usage'].get('total_tokens'):
                print(f'\nTokens: {response["usage"]["total_tokens"]}')
    
    # Save the conversation
    conversation.add_round(prompt, responses)
    await conversation.save()
    print('\n\nConversation saved to test-conversation.json')
    
    # Test assessment
    print('\n\n--- Running Assessment ---\n')
    
    assessment_prompt = conversation.format_assessment_prompt(prompt, responses)
    assessment_messages = [{'role': 'user', 'content': assessment_prompt}]
    
    assessment_responses = await call_models_in_parallel(models, assessment_messages)
    
    for model, response in assessment_responses.items():
        print(f'\n=== {model} Assessment ===')
        if 'error' in response:
            print(f'Error: {response["error"]}')
        else:
            print(response['content'][:500] + '...')
    
    conversation.add_round(assessment_prompt, assessment_responses, True)
    await conversation.save()


if __name__ == '__main__':
    asyncio.run(main())